### ISSUE TITLE
Fixing the visual encoders in RL baselines - Training policy only

### ISSUE THREAD

**Issue by yuyecreus** [facebookresearch/habitat-lab#650]
Hello everybody,

So I am experimenting with habitat baselines and I want to test some experiments on ImageNav (scene dataset is from Matterport3d).

I have fixed 1 apartment only (I am not using the default dataset for ImageNav which involves several apartments but only the dataset for my chosen apartment). For that I have implemented the adaption as mentioned in issue https://github.com/facebookresearch/habitat-lab/issues/636#issuecomment-832149590 . 

Now, when I run the default ResNet policy (which learns 2  ResNet encoders on-line for both goals and current observations) the training seems to work well as the rewards keep increasing and the losses take reasonable values. (i.e. action loss between -0.5 and 0.5, value loss between 0 and 1.2, dist_entropy goes from 1.2 in the beginning to 0.2 or so while training).

However, within the same-set up I am experimenting with a custom convolutional encoder that I have pre-trained in my chosen apartment. I use the Baseline policy (not the ResNet one) for loading my encoder (both for goals and current observations) and freezing its parameters. Now, in the RL training the action and value losses take values very close to 0 (i suppose because I am not allowing PPO to tune the encoders) but the policy does not learn correctly. (i.e. action loss between 0 and 0.03, value loss between 0 and 0.015, dist_entropy remains almost the same during training: between 1.34 and 1.38)

I would like to know if when fixing the visual encoders, there exists any fix to take into account for learning only the policy from scratch and adapting the PPO losses for that.

Thank you very much!

**Comment by erikwijmans** [comment]
There isn't anything obvious to change and in my experience the same setup works for training vs. frozen CNN.  I'd look for some bug in the loading code.
